corn-maze.lic

```diff
@@ -0,0 +1,632 @@
+    @full_path.each do |x|
+      exit if Flags['maze_done']
+      if x == 'loot'
+        wait_on_done_move
+        loot_room
+      else
+        custom_move("#{x}",true)
+      end
+    end
+    finish_task
+  end
+
+  def loot_room
+    pause 0.1 until XMLData.room_count != @last_looted
+    targets = DRRoom.room_objs.grep(/rake|stones|hay|wood|basket|boulder|wheel|spiderweb|straw|statue|pail|fence|wagon|burrow|hut/)
+    lootables = DRRoom.room_objs.grep(/hip pouch|pass|(agonite|steelsilk|loimic|dragonar|faeweave) scarecrow/)
+    lootables.each do |item|
+      stow_thing(item)
+    end
+    targets.each do |target|
+      if /manage/ =~ DRC.bput("search #{target.split.last}", "Sadly, you don't find anything", 'You manage to find', 'already been picked clean', 'I could not find what you were referring to', 'not in any condition', 'nothing of special interest about it', 'You ponder that')
+        while thing = DRC.left_hand || DRC.right_hand
+          if @looting
+            case thing
+            when *@junk_list
+              drop_item(thing)
+            else
+              stow_thing(thing)
+            end
+          elsif DRC.left_hand.include?('pass') || DRC.right_hand.include?('pass')
+            stow_thing(thing)
+          else
+            drop_item(thing)
+          end
+        end
+      end
+    end
+    @last_looted = XMLData.room_count
+  end
+
+  def drop_item(item)
+    case bput("drop my #{item}", 'You drop', 'would damage it', 'smashing it to bits', 'Something appears different about')
+    when 'would damage it', 'Something appears different about'
+      drop_item(item)
+    end
+  end
+
+  def wait_on_done_move
+    last_room = XMLData.room_count
+    pause 0.5
+    while last_room != XMLData.room_count
+      last_room = XMLData.room_count
+      pause 0.5
+    end
+  end
+
+  def get_task
+    wait_on_done_move
+    @halfling_path.each do |x|
+      exit if Flags['maze_done']
+      if DRRoom.npcs.include? 'Halfling'
+        break
+      else
+        custom_move "#{x}"
+      end
+    end
+    fput 'ask halfling about task'
+    echo "Task count = #{@task_count}"
+    if @searching == false
+      case bput('ask halfling for task', 'A cheerful Halfling smiles happily', 'To whom are you', 'A cheerful Halfling looks at you and says')
+      when 'A cheerful Halfling looks at you and says'
+        fput('ask halfling for task cancel')
+        fput('ask halfling for task cancel')
+        return get_task
+      end
+    end
+    @done = false
+    find_start
+    @done = false
+    wait_for_command
+  end
+
+  def find_start
+    wait_on_done_move
+    unless XMLData.room_title.include?('Labyrinth') || XMLData.room_title.include?('Corn Maze, Entrance')
+      DRC.message("The script can't find its way back from here!  Make your way back towards the northwest Labyrinth using elanthipedia's Corn Maze entry, such as https://elanthipedia.play.net/Sleeping_Dragon_Corn_Maze_432 and then unpause the script!")
```

Comment:
What would you think about saying starting narost right clicking and picking the corn maze map here instead?

Reply:
Yeah, I didn't know that was an option, but it's a lot easier.